source/octf/utils/ShellExecutor.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright(c) 2012-2018 Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause-Clear
+ */
+
+#include <octf/utils/ShellExecutor.h>
+#include <octf/utils/Exception.h>
+#include <cstdio>
+#include <memory>
+#include <array>
+
+namespace octf {
+
+
+std::string ShellExecutor::executeCommand() {
+	std::string command = m_command;
+	command += " ";
+	command += m_args;
+
+	const char* cmd = command.c_str();
+
+	std::array<char, 128> buffer;
+	std::string result;
+
+	std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd, "r"), pclose);
+	if (!pipe) {
+		throw Exception("Call to popen() failed!");
+	}
+
+	int read;
+	while ((read = fread(buffer.data(), 1, buffer.size(), pipe.get()))
+			== static_cast<int>(buffer.size())) {
```

Comment:
Do you need cast here?

Reply:
Without a cast there is an unsigned with signed comparison. It does not matter now with current values but the buffer size may be changed in the future, and also the compiler warnings are suppressed.